src/test/java/bio/terra/service/common/azure/AzureUriUtilsTest.java

```diff
@@ -0,0 +1,40 @@
+package bio.terra.service.common.azure;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import com.azure.storage.blob.BlobUrlParts;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles({"google", "unittest"})
+@Tag("bio.terra.common.category.Unit")
+class AzureUriUtilsTest {
+
+  private static Stream<Arguments> testEncodedUrls() {
+    return Stream.of(
+        arguments(
+            BlobUrlParts.parse("https://foo.blob.core.windows.net/container/blobname.txt"),
```

Comment:
To be safe, I would call `BlobUrlParts.parse()` inside the test in case it throws. Also it looks like the strings are identical, if so you could simplify by passing only the URL string to the test case.